src/Abstractions/RetryPolicy.cs

```diff
@@ -123,11 +126,56 @@ public class RetryPolicy
     /// </value>
     public TimeSpan RetryTimeout { get; }
 
+    /// <summary>
+    /// Gets a delegate to call on exception to determine if retries should proceed.
+    /// </summary>
+    /// <value>
+    /// Defaults delegate that always returns true (i.e., all exceptions are retried).
+    /// </value>
+    public Func<Exception, bool> Handle { get; private init; }
+
 #pragma warning disable SA1623 // Property summary documentation should match accessors
     /// <summary>
     /// This functionality is not implemented. Will be removed in the future. Use TaskOptions.FromRetryHandler instead.
     /// </summary>
     [Obsolete("This functionality is not implemented. Will be removed in the future. Use TaskOptions.FromRetryHandler instead.")]
```

Comment:
```suggestion
    [Obsolete("This functionality is not implemented. Will be removed in the future. Use TaskOptions.FromRetryHandler or HandleTaskFailureDetails instead.")]
```

Reply:
Because I added the Obsolete attribute, the code generates the following warning twice:

`'RetryPolicy.Handle' is obsolete: 'This functionality is not implemented. Will be removed in the future. Use TaskOptions.FromRetryHandler or HandleFailure instead.'	Abstractions	C:\DevOps\durabletask-dotnet\src\Abstractions\RetryPolicy.cs	163`

I have added the suppression of this warning around the two statements that set `Handle` so no warning appear.

Reply:
May have been some miscommunication: can we add the ` or HandleFailure` to this existing `[Obsolete]` attribute? No need to add the attribute to the new `Handle` property. Because it has `private init;` it is well-enough protected.

Reply:
Corrected.